ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/RetryingGetStorageRangeFromPeerTask.java

```diff
@@ -44,8 +43,13 @@ private RetryingGetStorageRangeFromPeerTask(
       final Bytes32 startKeyHash,
       final Bytes32 endKeyHash,
       final BlockHeader blockHeader,
-      final MetricsSystem metricsSystem) {
-    super(ethContext, 4, data -> data.proofs().isEmpty() && data.slots().isEmpty(), metricsSystem);
+      final MetricsSystem metricsSystem,
+      final int maxRetries) {
+    super(
+        ethContext,
+        metricsSystem,
+        data -> data.proofs().isEmpty() && data.slots().isEmpty(),
```

Comment:
A bad peer will be a peer with this kind of result
I’m not sure you are checking if the response is like that in order to switch to another peer

Reply:
check this other PR https://github.com/hyperledger/besu/pull/5509, that checks and records the empty response in the `AbstractRetryingPeerTask`